src/engine/GroupBy.cpp

```diff
@@ -530,20 +526,27 @@ cppcoro::generator<IdTable> GroupBy::computeResultLazily(
     groupSplitAcrossTables = true;
     lazyGroupBy.processBlock(evaluationContext, lastBlockStart, idTable.size());
     if (!singleIdTable && !resultTable.empty()) {
-      co_yield resultTable;
+      currentLocalVocab.mergeWith(storedLocalVocabs);
+      Result::IdTableVocabPair outputPair{std::move(resultTable),
+                                          std::move(currentLocalVocab)};
+      co_yield outputPair;
+      // Reuse buffer if not moved out
+      resultTable = std::move(outputPair.idTable_);
       resultTable.clear();
+      currentLocalVocab = LocalVocab{};
+      storedLocalVocabs.clear();
```

Comment:
Okay, Each time you reset the `currentGroupBlock` you have to keep the corresponding local vocab around and make it part of the result that performs the corresponding `commitRow`.
Either explicitly store the pair of currentGroupBlock + its local vocab, or even rewrite the current group block 
(find all local vocab entries, and copy them to a new local vocab and rewrite the IDs.